vipermonkey/core/expressions.py

```diff
@@ -2088,12 +2109,19 @@ def _handle_regex_execute(self, context, tmp_lhs):
             return None
 
         # Find all the regex matches in the string.
-        r = None
         try:
-            r = re.findall(pat, mod_str)
+            matches = re.finditer(pat, mod_str)
+            match_objects = []
+            for match in matches:
+                # create RegexResult so that ViperMonkey can access fields like FirstIndex
+                match_object = RegexResult(str(match), int(match.start()))
+                match_objects.append(match_object)
+            return match_objects
         except Exception as e:
             log.error("Regex.Execute() failed. " + safe_str_convert(e))
-        return r
+            return None
+
+        return match_objects
```

Comment:
This line should never run, either there is an exception and we've returned None, or we've returned match_objects from the try block already.